### PR TITLE
feature/issue 1731

### LINKED ISSUE #1731: mark binomial_exact function for removal
**What should be changed**

`cogent3.maths.stats.binomial_exact()`

**Why**

Not used in cogent3, available in scipy.stats?

**How much notice**

6 months notice. This should be represented in the user warning as a specific version. e.g. <current year>.<current month + *x*> where *x* is from the PR date.  

**Guidelines for deprecation**

As this is a function that will be removed, see [dev guidelines for deprecation](https://github.com/cogent3/c3dev/wiki/Deprecating-Code) of code. Indicate the limiting version (for when the old function will be removed) as 3 months from when you make the change.

### PR DESCRIPTION
**Pull request by ChangsenJiang**
DEP: remove the binomial_exact() function and the related tests , fixes #1731  
Remove the `cogent3.maths.stats.binomial_exact()` function, and removed the related `test_binomial_series()`,  `test_binomial_exact()`, `test_binomial_exact_floats()`, and `test_binomial_exact_errors()`

### CODE REVIEW

File: src/cogent3/maths/stats/distribution.py [hunk 1]
```diff
@@ -72,20 +73,12 @@ def poisson_exact(successes, mean):
         return pdtrc(successes - 1, mean) - pdtrc(successes, mean)
 
 
+@c3warn.deprecated_callable(
+    version="2024.9", reason="use scipy.stats.binom.pmf()instead", is_discontinued=True
+)
 def binomial_exact(successes, trials, prob):
```
Comment (GavinHuttley) [inline src/cogent3/maths/stats/distribution.py:79]: The point here is not to replace our function with that from Scipy; rather, just use the decorator as indicated. Revert the change within the function to the original.

File: src/cogent3/maths/stats/distribution.py [hunk 2]
```diff
@@ -5,7 +5,7 @@
 
 from numpy import arctan as atan
 from numpy import array, exp, sqrt
-from scipy.stats import f, norm, t
+from scipy.stats import binom, f, norm, t
```
Comment (GavinHuttley) [inline src/cogent3/maths/stats/distribution.py:8]: delete the new import

**Review by GavinHuttley — CHANGES_REQUESTED** [review]
Thanks for the contribution!

When marking a function to be discontinued we limit our changes to applying the decorator and giving the reason (including referring users to a replacement in another project). In other words, we do not actually change our code within the function since we are going to be deleting it.

File: src/cogent3/maths/stats/distribution.py (hunk 1 above)
Comment (ChangsenJiang) [reply src/cogent3/maths/stats/distribution.py:79]: Thanks for your reply! Does that mean what I'm going to do is keep the @c3warn.deprecated_callable message at the top, revert everything inside the function (including return), and then delete the new modules I imported?

File: src/cogent3/maths/stats/distribution.py (hunk 2 above)
Comment (ChangsenJiang) [reply src/cogent3/maths/stats/distribution.py]: Now I revert the change within the function to the original and delete the new imports. Please have a look : )

**Comment by ChangsenJiang** [comment]
Now I revert the change within the function to the original and delete the new imports. Please have a look : )

**Comment by ChangsenJiang** [comment]
Now I revert the change within the function to the original and delete the new imports. Please have a look : )

**Review by GavinHuttley — CHANGES_REQUESTED** [review]
Thanks for your effort, but I can't accept this PR as is.

First, the replacement function in scipy you have identified is incorrect. It should be `scipy.stats.binomtest`. This means you would need to update your deprecation reason and, in doing so, correct the spacing between words.

Second, you would need to identify all uses of `binomial_exact()` in `cogent3` and replace them with correct usage of the scipy function.

Finally, be sure and delete imports of `binomial_exact()` since it is no longer being used.

**Comment by GavinHuttley** [comment]
Closing due to lack of response to review. But thanks for your effort!
